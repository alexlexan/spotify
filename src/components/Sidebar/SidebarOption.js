import React from "react";
import "./SidebarOption.sass";
import { NavLink } from "react-router-dom";
import { useDispatch } from "react-redux";
import { setActivePlaylist } from "../../store/actions";

function SidebarOption({ option = "test", Icon, id, playlist }) {
  const dispatch = useDispatch();
  const getPlaylists = (item) => {
    dispatch(setActivePlaylist(item));
  };

  return (
    <div className="sidebarOption">
      {Icon && <Icon className="sidebarOption__icon" />}
      {Icon ? (
        <h4>{option}</h4>
      ) : (
        <NavLink
          to={`/playlists/${id}/sidebarPlaylist`}
          activeClassName="sidebarOptionActive"
          onClick={() => getPlaylists(playlist)}
        >
          <p>{option}</p>
        </NavLink>
      )}
    </div>
  );
}

export default SidebarOption;
