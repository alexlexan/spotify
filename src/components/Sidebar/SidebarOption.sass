.sidebarOption 
  display: flex
  cursor: pointer
  align-items: center
  height: 40px
  color: grey
  transition: 200ms color ease-in
  a 
    text-decoration: none
    color: grey
    padding: 0 12px
    transition: 200ms color ease-in
    &:hover
      color: white
      transition: 200ms color ease-in

.sidebarOption:hover 
  color: white

.sidebarOptionActive
  color: white    


.sidebar > hr 
  border: 1px solid #282828
  width: 90%
  margin: 10px auto


.sidebarOption__icon 
  padding-left: 10px
  padding-right: 10px


.sidebarOption > p 
  margin-top: 10px
  margin-left: 20px
  font-size: 14px
