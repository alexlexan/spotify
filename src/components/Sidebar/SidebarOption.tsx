import { SvgIconTypeMap } from "@material-ui/core";
import { OverridableComponent } from "@material-ui/core/OverridableComponent";
import React from "react";
import "./SidebarOption.sass";

type Props = {
  option: string;
  Icon: OverridableComponent<SvgIconTypeMap<{}, "svg">>;
};

const SidebarOption: React.FC<Props> = ({ option = "test", Icon }) => {
  return (
    <div className="sidebarOption">
      <Icon className="sidebarOption__icon" />
      <h4>{option}</h4>
    </div>
  );
};

export default SidebarOption;
