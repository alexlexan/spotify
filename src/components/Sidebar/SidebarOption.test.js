import React from "react";
import SidebarOption from "./SidebarOption.tsx";
import HomeIcon from "@material-ui/icons/Home";

test("render SidebarOptionOption component", () => {
  const component = shallow(<SidebarOption Icon={HomeIcon} />);
  expect(component).toMatchSnapshot();
});
