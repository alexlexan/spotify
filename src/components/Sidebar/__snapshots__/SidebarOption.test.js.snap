// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SidebarOptionOption component 1`] = `
<div
  className="sidebarOption"
>
  <HomeIcon
    className="sidebarOption__icon"
  />
  <h4>
    test
  </h4>
</div>
`;
